build/charts/theia/values.yaml

```diff
@@ -0,0 +1,101 @@
+    # Vary from 0 to 1.
+    threshold: 0.5
+    # -- The percentage of records in ClickHouse that will be deleted when the
+    # storage grows above threshold. Vary from 0 to 1.
+    deletePercentage: 0.5
+    # -- The time interval between two round of monitoring. Can be a plain integer using
+    # one of these unit suffixes ns, us (or µs), ms, s, m, h.
+    execInterval: "1m"
+    # -- The number of rounds for the monitor to stop after a deletion to wait for
+    # the ClickHouse MergeTree Engine to release memory.
+    skipRoundsNum: 3
+    # -- Container image to use for the ClickHouse Monitor.
+    image:
+      repository: "projects.registry.vmware.com/antrea/theia-clickhouse-monitor"
+      pullPolicy: "IfNotPresent"
+      tag: "latest"
+  secret:
+    # -- ClickHouse username. It will be stored in a secret.
+    username: "clickhouse_operator"
+    # -- ClickHouse password. It will be stored in a secret.
+    password: "clickhouse_operator_password"
+  service:
+    # -- The type of service exposes ClickHouse. It can be one of ClusterIP, NodePort or LoadBalancer.
+    type: ClusterIP
+    # -- TCP port number for the ClickHouse service.
+    tcpPort: 9000
+    # -- TCP port number for the ClickHouse service.
+    httpPort: 8123
+  # -- ClickHouse storage size. Can be a plain integer or as a fixed-point
+  # number using one of these quantity suffixes: E, P, T, G, M, K. Or the
+  # power-of-two equivalents: Ei, Pi, Ti, Gi, Mi, Ki.
+  storageSize: "8Gi"
+  # -- Time to live for data in the ClickHouse. Can be a plain integer using
+  # one of these unit suffixes SECOND, MINUTE, HOUR, DAY, WEEK, MONTH, QUARTER,
+  # YEAR.
+  ttl: 1 HOUR
+  persistentVolume:
+    # -- Enable deploying the ClickHouse with Persistent Volumes.
+    enable: false
+    # -- Persistent Volume Provisioner. Required if Persistent Volumes is enable.
+    # It must be one of "StorageClass", "Local", "NFS".
+    provisioner: "Local"
+    # -- The local path. Required when Persistent Volumes is enable and the
+    # provisioner is "Local".
+    localPath: "/data"
+    # -- Affinity for the Local Persistent Volume. Required when Persistent
+    # Volumes is enable and the provisioner is "Local". Please make sure the
+    # Node used to store the ClickHouse data satisfy the affinity requirements.
+    affinity: 
+      required:
+        nodeSelectorTerms:
+        - matchExpressions:
+          - key: antrea.io/clickhouse-data-node
+            operator: Exists
+    # -- The NFS server hostname or IP address. Required when Persistent Volumes
+    # is enable the provisioner is "NFS".
+    nfsHost: ""
+    # -- The path exported on the NFS server. Required when Persistent Volumes
+    # is enable the provisioner is "NFS".
+    nfsPath: ""
+    # -- The StorageClass used to dynamically provision the Persistent Volume.
+    # Required when Persistent Volumes is enable the provisioner is "StorageClass".
+    storageClass: ""
+grafana:
```

Comment:
have we considered the case where a user is already running an instance of Grafana and want to use the same instance for flow visibility? @salv-orlando

Reply:
I think we do not have a design for this before. To my knowledge, it is possible to connect an existing Grafana to ClickHouse database by setting the service IP and credentials in the Grafana UI. 

But If the existing instance of Grafana is out of cluster, it might have trouble to connect to ClickHouse as ClickHouse is exposed with a ClusterIP service.

And it seems users have to take care of the dashboards import and plugins installation by themselves. I'm not sure whether it is convenient for the users to do these. 

Also would like to include @heanlan in this discussion. Please correct me if I'm wrong.

Reply:
@antoninbas I think the answer is that so far we have only been considering a scenario where antrea network flow visibility has its own grafana instance.

But to the best of my knowledge there is nothing special in the way we deploy grafana. In theory we could have users leverage their own grafana instance. I would not know however how to automate dashboard provisioning with helm in this case. This is something we can considering for a future release.

Reply:
I understand that this is not something we need to look at in details now. I just wanted to make sure that this is something that we will be able to accomodate without any significant change to the Helm values file. For example, we should be able to easily disable the deployment of Grafana (e.g., by setting `grafana.enable` to `false`). I don't know if anything else would be required that may have an impact on the structure of values.yaml.

Reply:
As ClickHouse does not have dependency on Grafana, I think disabling Grafana will not require a significant change to the Helm values file. There might require some extra steps for user to update their own Grafana instance to display the dashboards, but these should not be related to our Helm chart.